### ISSUE TITLE
[TPU] Can't launch a TPU Pod

### ISSUE THREAD

**Issue by infwinston** [skypilot-org/skypilot#1480]
Now launching a TPU Pod will trigger below error
```
  File "/Users/weichiang/repos/sky/sky/utils/common_utils.py", line 233, in _record
    return f(*args, **kwargs)
  File "/Users/weichiang/repos/sky/sky/backends/cloud_vm_ray_backend.py", line 1345, in _gang_schedule_ray_up
    self._tpu_pod_setup(cluster_config_file, cluster_handle)
  File "/Users/weichiang/repos/sky/sky/backends/cloud_vm_ray_backend.py", line 1161, in _tpu_pod_setup
    if len(all_ips) != num_tpu_devices:
TypeError: object of type 'NoneType' has no len()
```

This seems to be introduced after https://github.com/skypilot-org/skypilot/pull/1380 where 
https://github.com/skypilot-org/skypilot/blob/5506e1dec428c8bd6dff4110ea86e6c5e76bd9dd/sky/backends/cloud_vm_ray_backend.py#L1158
returns an empty list and caused the bug.

@iojw Should the semantic of `handle.external_ips()` always returns a list of IPs?

Also we should probably make TPU Pod's smoke test run by default to avoid such bug.

**Comment by iojw** [comment]
@infwinston Thank you for catching this! Yes you are right, I think it should return an empty list if there is no head node. I will fix this.

**Comment by Michaelvll** [comment]
Ahh, good catch! We should enable the TPU pod smoke test as that cannot be covered by any other tests.

I think the problem is we should use `use_cached_ips=False` here, since it is before the handle is set up.

**Comment by iojw** [comment]
Closed by #1481.
